internal/controller/scheduler/scheduler.go

```diff
@@ -1065,13 +1050,22 @@ func (w *worker) createWorkspaceSetupContainer(podSpec *corev1.PodSpec, workspac
 	default:
 		securityContext = nil
 	}
+
 	// Init containers. These run in order before the regular containers.
 	// We run some init containers before any specified in the given podSpec.
 	//
-	// We use an init container to copy buildkite-agent into /workspace.
+	// The very first init container prepares the workspace:
+	// - chown the workspace directory according to the security context
+	// - make a sockets directory usable by any user
+	// - copy buildkite-agent and tini-static into /workspace
```

Comment:
[kinda related] I wonder if we should use `exec` instead of subprocess in `kubernetes-bootstrap` command in agent.